perf/perf_rawevents.py

```diff
@@ -88,18 +85,21 @@ def error_check(self):
                 self.log.info("Failed command: %s", self.fail_cmd[cmd])
             self.fail("perf_raw_events: refer log file for failed events")
 
-    def test(self):
-        cpu_family = cpu.get_family()
-        if cpu_family == 'power8':
-            self.run_event('raw_codes_p8', 'raw')
-            self.error_check()
-            self.run_event('name_events_p8', 'name')
-            self.error_check()
-        elif cpu_family == 'power9':
-            self.run_event('raw_codes_p9', 'raw')
-            self.error_check()
-        else:
-            self.cancel('This test is not supported on %s' % cpu_family)
+    def test_raw_code(self):
+        cpu_family = cpu.get_family()[5:]
+        event = 'raw'
+        file_name = event + '_codes_p' + cpu_family
```

Comment:
@disgoel Good implementation.
But you can still reduce the duplicates and optimize code as below,
make "cpu_family"  variable as global so that you will end up calling cpu.get_family() only once instead of twice.
so move this under setUp() as below
`self.cpu_family = cpu.get_family()[5:]`

Next no need of variable event here as this function is specifically for raw_code only.
So this can be as below
`file_name = raw_codes_p + self.cpu_family`

Reply:
@Naresh-ibm 

localhost:/home/avocado-fvt-wrapper/tests/avocado-misc-tests/perf # avocado run --test-runner runner perf_rawevents.py
JOB ID     : 0b5c18dbb8f7e9ba04f1710e9e36f2a58c869d8d
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-03-21T17.18-0b5c18d/job.log
 (1/2) perf_rawevents.py:PerfRawevents.test_raw_code: PASS (266.58 s)
 (2/2) perf_rawevents.py:PerfRawevents.test_name_event: PASS (266.12 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-03-21T17.18-0b5c18d/results.html
JOB TIME   : 542.21 s